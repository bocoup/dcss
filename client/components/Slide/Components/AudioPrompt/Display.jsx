import { type } from './meta';
import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Container, Header, Segment } from '@components/UI';
import PromptRequiredLabel from '../PromptRequiredLabel';
import ResponseRecall from '@components/Slide/Components/ResponseRecall/Display';
import AudioRecorder from '@components/Slide/Components/AudioPrompt/AudioRecorder';
import { getResponse } from '@actions/response';
import '@components/Slide/Components/AudioPrompt/AudioPrompt.css';

class Display extends Component {
  constructor(props) {
    super(props);

    const { persisted = { value: '', transcript: '' } } = this.props;

    this.state = {
      isReady: false,
      autostart: false,
      playing: this.isScenarioRun,
      transcript: persisted.transcript,
      type: '',
      value: persisted.value
    };

    this.created_at = new Date().toISOString();
    this.onChange = this.onChange.bind(this);
  }

  get isScenarioRun() {
    return location.pathname.includes('/run/');
  }

  async componentDidMount() {
    if (!this.isScenarioRun) {
      this.setState({
        isReady: true
      });
      return;
    }
    let {
      getResponse,
      onResponseChange,
      persisted = {},
      responseId,
      run
    } = this.props;

    let { name = responseId, transcript = '', value = '' } = persisted;

    let state = {
      ...this.state
    };

    if (!value || !transcript) {
      const previous = await getResponse({
        id: run.id,
        responseId
      });

      if (previous && previous.response) {
        value = previous.response.value;
        transcript = previous.response.transcript;
      }
    }

    if (value) {
      onResponseChange({}, { name, transcript, value, isFulfilled: true });
      state = {
        ...state,
        transcript,
        value
      };
    }

    this.setState({
      ...state,
      isReady: true
    });
  }

  onChange(event, { name, value }) {
    const { created_at } = this;
    const { recallId } = this.props;
    const transcript = '';
    this.props.onResponseChange(event, {
      created_at,
      ended_at: new Date().toISOString(),
      recallId,
      name,
      transcript,
      type,
      value
    });

    this.setState({ transcript, value });
  }

  render() {
    const {
      isReady,
      autostart,
      isRecording,
      playing,
      transcript,
      value
    } = this.state;

    if (!isReady) {
      return null;
    }

    const {
      prompt,
      recallId,
      responseId,
      required,
      run,
    } = this.props;
    const { onChange } = this;
    const isFulfilled = value ? true : false;
    const header = (
      <Fragment>
        {prompt} {required && <PromptRequiredLabel fulfilled={isFulfilled} />}
      </Fragment>
    );

    const willShowAudioRecorder = isFulfilled || autostart;

    return (
      <Fragment>
        <Segment>
          <Header as="h3">{header}</Header>
          {recallId && <ResponseRecall run={run} recallId={recallId} />}
          <AudioRecorder
            autostart={autostart}
            getResponse={this.props.getResponse}
            isEmbeddedInSVG={this.props.isEmbeddedInSVG}
            isRecording={isRecording}
            onChange={onChange}
            prompt={prompt}
            responseId={responseId}
            run={run}
            transcript={transcript}
            value={value}
          />
        </Segment>
      </Fragment>
    );
  }
}

Display.defaultProps = {
  isEmbeddedInSVG: false
};

Display.propTypes = {
  isEmbeddedInSVG: PropTypes.bool,
  getResponse: PropTypes.func,
  isRecording: PropTypes.bool,
  onResponseChange: PropTypes.func,
  persisted: PropTypes.object,
  placeholder: PropTypes.string,
  prompt: PropTypes.string,
  recallId: PropTypes.string,
  required: PropTypes.bool,
  responseId: PropTypes.string,
  run: PropTypes.object,
  type: PropTypes.oneOf([type]).isRequired
};

const mapStateToProps = state => {
  const { run } = state;
  return { run };
};

const mapDispatchToProps = dispatch => ({
  getResponse: params => dispatch(getResponse(params))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Display);
