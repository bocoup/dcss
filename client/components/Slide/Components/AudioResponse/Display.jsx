import { type } from './meta';
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
  Button,
  Form,
  Header,
  Icon,
  Message,
  Segment
} from 'semantic-ui-react';
import PromptRequiredLabel from '../PromptRequiredLabel';
import ResponseRecall from '@components/Slide/Components/ResponseRecall/Display';
import MicRecorder from 'mic-recorder-to-mp3';
import { detect } from 'detect-browser';
import { connect } from 'react-redux';
import { getResponse } from '@client/actions/response';
import './AudioResponse.css';

const SUPPORTED_BROWSERS = ['chrome', 'firefox'];

class Display extends Component {
  constructor(props) {
    super(props);

    const { persisted = { value: '', transcript: '' } } = this.props;

    this.state = {
      blobURL: '',
      isRecording: false,
      transcript: persisted.transcript,
      type: '',
      value: persisted.value
    };

    this.created_at = new Date().toISOString();
    this.mp3Recorder = new MicRecorder({ bitRate: 128 });

    this.onChange = this.onChange.bind(this);
    this.onFocus = this.onFocus.bind(this);
    this.onStart = this.onStart.bind(this);
    this.onStop = this.onStop.bind(this);

    this.browserSupported = SUPPORTED_BROWSERS.includes(detect().name);
  }

  get isScenarioRun() {
    return location.pathname.includes('/run/');
  }

  async componentDidMount() {
    if (!this.isScenarioRun) {
      return;
    }
    let {
      getResponse,
      onResponseChange,
      persisted = {},
      responseId,
      run
    } = this.props;

    let { name = responseId, transcript = '', value = '' } = persisted;

    if (!value || !transcript) {
      const previous = await getResponse({
        id: run.id,
        responseId
      });

      if (previous && previous.response) {
        value = previous.response.value;
        transcript = previous.response.transcript;
      }
    }

    if (value) {
      onResponseChange({}, { name, transcript, value, isFulfilled: true });
      this.setState({ transcript, value });
    }
  }

  async onStart() {
    await this.mp3Recorder.start();
    this.created_at = new Date().toISOString();
    this.setState({
      isRecording: true
    });
  }

  async onStop() {
    const [buffer, blob] = await (await this.mp3Recorder.stop()).getMp3();
    const blobURL = URL.createObjectURL(blob);

    const file = new File(buffer, 'recording.mp3', {
      type: blob.type,
      lastModified: Date.now()
    });

    const { responseId, responseId: name, run } = this.props;

    let body = new FormData();
    body.append('name', 'audio-response');
    body.append('recording', file);
    body.append('responseId', responseId);

    if (run) {
      body.append('runId', run.id);
    }

    const { s3Location: value } = await (await fetch('/api/media/audio', {
      method: 'POST',
      body
    })).json();

    this.setState(prevState => {
      if (prevState.blobURL) {
        URL.revokeObjectURL(prevState.blobURL);
      }
      return { blobURL, value, isRecording: false };
    });

    const { created_at } = this;
    const { recallId } = this.props;
    const transcript = '';

    this.props.onResponseChange(
      {},
      {
        created_at,
        ended_at: new Date().toISOString(),
        name,
        recallId,
        transcript,
        type,
        value
      }
    );

    this.setState({ transcript, value });
  }

  onFocus() {
    if (!this.created_at) {
      this.created_at = new Date().toISOString();
    }
  }

  onChange(event, { name, value }) {
    const { created_at } = this;
    const { recallId } = this.props;
    const transcript = '';
    this.props.onResponseChange(event, {
      created_at,
      ended_at: new Date().toISOString(),
      recallId,
      name,
      transcript,
      type,
      value
    });

    this.setState({ transcript, value });
  }

  render() {
    const { isRecording, blobURL, transcript, value } = this.state;
    const { prompt, recallId, responseId, required, run } = this.props;
    const { onChange, onFocus } = this;
    const fulfilled = value || blobURL ? true : false;
    const header = this.browserSupported ? (
      required && <PromptRequiredLabel fulfilled={fulfilled} />
    ) : (
      <React.Fragment>
        {prompt} {required && <PromptRequiredLabel fulfilled={fulfilled} />}
      </React.Fragment>
    );

    return this.browserSupported ? (
      <Segment>
        <Header as="h3">{header}</Header>
        {recallId && <ResponseRecall run={run} recallId={recallId} />}
        {!isRecording && (
          <Button basic toggle onClick={this.onStart}>
            <Icon name="circle" aria-label="Record an Audio Response" />
            {prompt}
          </Button>
        )}
        {isRecording && (
          <Button basic negative onClick={this.onStop}>
            <Icon name="stop circle" aria-label="Record an Audio Response" />
            Done
          </Button>
        )}

        {(blobURL || value) && (
          <Message
            content={
              <React.Fragment>
                <audio
                  src={blobURL || `/api/media/${value}`}
                  controls="controls"
                />

                <blockquote className="audiotranscript__blockquote">
                  {transcript ||
                    '(Transcription in progress. This may take a few minutes, depending on the length of your audio recording.)'}
                </blockquote>
              </React.Fragment>
            }
          />
        )}
      </Segment>
    ) : (
      <Segment>
        <Header as="h3">{header}</Header>

        <Form>
          <Form.TextArea
            name={responseId}
            placeholder="..."
            onFocus={onFocus}
            onChange={onChange}
          />
        </Form>
      </Segment>
    );
  }
}

Display.propTypes = {
  getResponse: PropTypes.func,
  isRecording: PropTypes.bool,
  onResponseChange: PropTypes.func,
  persisted: PropTypes.object,
  placeholder: PropTypes.string,
  prompt: PropTypes.string,
  recallId: PropTypes.string,
  required: PropTypes.bool,
  responseId: PropTypes.string,
  run: PropTypes.object,
  type: PropTypes.oneOf([type]).isRequired
};

const mapStateToProps = state => {
  const { run } = state;
  return { run };
};

const mapDispatchToProps = dispatch => ({
  getResponse: params => dispatch(getResponse(params))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Display);
