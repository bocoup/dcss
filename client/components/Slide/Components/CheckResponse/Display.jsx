import { type } from './type';
import React from 'react';
import PropTypes from 'prop-types';
import { Label, Checkbox } from 'semantic-ui-react';

import './CheckResponse.css';

const Display = ({
    prompt,
    checked,
    unchecked,
    responseId,
    onResponseChange
}) => (
    <React.Fragment>
        <Label as="label" className="CheckResponse-label">
            <Checkbox
                name={responseId}
                value={checked}
                label={prompt}
                onChange={(event, values) => {
                    if (values.checked) {
                        values.value = checked;
                    } else {
                        values.value = unchecked;
                    }
                    if (onResponseChange) onResponseChange(event, values);
                }}
                className="CheckResponse-input"
            />
        </Label>
    </React.Fragment>
);

Display.propTypes = {
    prompt: PropTypes.string,
    checked: PropTypes.string,
    unchecked: PropTypes.string,
    responseId: PropTypes.string,
    onResponseChange: PropTypes.func,
    type: PropTypes.oneOf([type]).isRequired
};

export default React.memo(Display);
