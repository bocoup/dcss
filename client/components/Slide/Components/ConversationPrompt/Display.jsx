import { type } from './meta';
import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Container, Header, Segment } from '@components/UI';
import PromptRequiredLabel from '../PromptRequiredLabel';
import ResponseRecall from '@components/Slide/Components/ResponseRecall/Display';
import AudioRecorder from '@components/Slide/Components/AudioPrompt/AudioRecorder';
import { getResponse } from '@actions/response';
import '@components/Slide/Components/AudioPrompt/AudioPrompt.css';
import './ConversationPrompt.css';

import Hls from 'hls.js';
import Player from 'react-player';

import { YOUTUBE_PLAYER_DISPLAY_VARS } from './constants';

window.Hls = Hls;

class Display extends Component {
  constructor(props) {
    super(props);

    const { persisted = { value: '', transcript: '' } } = this.props;

    this.state = {
      isReady: false,
      autostart: false,
      playing: this.isScenarioRun,
      transcript: persisted.transcript,
      type: '',
      value: persisted.value
    };

    this.created_at = new Date().toISOString();
    this.onChange = this.onChange.bind(this);
  }

  get isScenarioRun() {
    return location.pathname.includes('/run/');
  }

  async componentDidMount() {
    if (!this.isScenarioRun) {
      this.setState({
        isReady: true
      });
      return;
    }
    let {
      getResponse,
      onResponseChange,
      persisted = {},
      responseId,
      run
    } = this.props;

    let { name = responseId, transcript = '', value = '' } = persisted;

    let state = {
      ...this.state
    };

    if (!value || !transcript) {
      const previous = await getResponse({
        id: run.id,
        responseId
      });

      if (previous && previous.response) {
        value = previous.response.value;
        transcript = previous.response.transcript;
      }
    }

    if (value) {
      onResponseChange({}, { name, transcript, value, isFulfilled: true });
      state = {
        ...state,
        transcript,
        value
      };
    }

    this.setState({
      ...state,
      isReady: true
    });
  }

  onChange(event, { name, value }) {
    const { created_at } = this;
    const { recallId } = this.props;
    const transcript = '';
    this.props.onResponseChange(event, {
      created_at,
      ended_at: new Date().toISOString(),
      recallId,
      name,
      transcript,
      type,
      value
    });

    this.setState({ transcript, value });
  }

  render() {
    const {
      isReady,
      autostart,
      isRecording,
      playing,
      transcript,
      value
    } = this.state;

    if (!isReady) {
      return null;
    }

    const {
      prompt,
      configuration,
      recallId,
      responseId,
      required,
      run,
      url
    } = this.props;
    const { onChange } = this;
    const isFulfilled = value ? true : false;
    const header = (
      <Fragment>
        {prompt} {required && <PromptRequiredLabel fulfilled={isFulfilled} />}
      </Fragment>
    );

    const config = {
      youtube: {
        ...YOUTUBE_PLAYER_DISPLAY_VARS
      }
    };

    if (configuration.end) {
      config.youtube.playerVars.start = configuration.start;
      config.youtube.playerVars.end = configuration.end;
    }

    const light = !this.isScenarioRun;
    const height = '320px';
    const width = '100%';

    const ref = player => {
      this.player = player;
    };

    const onEnded = () => {
      this.setState({ autostart: true, playing: false });
      this.player.seekTo(configuration.start);
    };

    //
    //
    // TODO: support for "start" and "end" in other players
    // will be need to be manually set.
    //
    // const onProgress = ({playedSeconds}) => {
    //   console.log("duration: ", duration);
    //   console.log("playedSeconds: ", playedSeconds);
    //   console.log("triggerAt: ", triggerAt);
    //   if (configuration.kind === 'slice' && playedSeconds >= triggerAt) {
    //     this.setState({ autostart: true });
    //   } else {
    //     if (duration && playedSeconds >= duration - 0.5) {
    //       console.log("almost the end");
    //     }
    //   }
    // };
    //
    //

    const playerProps = {
      className: 'cp__video-player',
      config,
      light,
      height,
      onEnded,
      // onProgress,
      playing,
      ref,
      width,
      url
    };

    const willShowAudioRecorder = isFulfilled || autostart;

    return (
      <Fragment>
        <Container className="cp__video-wrapper">
          <Player {...playerProps} />
        </Container>
        {willShowAudioRecorder ? (
          <Segment>
            <Header as="h3">{header}</Header>
            {recallId && <ResponseRecall run={run} recallId={recallId} />}
            <AudioRecorder
              autostart={autostart}
              getResponse={this.props.getResponse}
              isEmbeddedInSVG={this.props.isEmbeddedInSVG}
              isRecording={isRecording}
              onChange={onChange}
              responseId={responseId}
              run={run}
              transcript={transcript}
              value={value}
            />
          </Segment>
        ) : null}
      </Fragment>
    );
  }
}

Display.defaultProps = {
  isEmbeddedInSVG: false
};

Display.propTypes = {
  configuration: PropTypes.object,
  isEmbeddedInSVG: PropTypes.bool,
  getResponse: PropTypes.func,
  isRecording: PropTypes.bool,
  onResponseChange: PropTypes.func,
  persisted: PropTypes.object,
  placeholder: PropTypes.string,
  prompt: PropTypes.string,
  recallId: PropTypes.string,
  required: PropTypes.bool,
  responseId: PropTypes.string,
  run: PropTypes.object,
  type: PropTypes.oneOf([type]).isRequired,
  url: PropTypes.string
};

const mapStateToProps = state => {
  const { run } = state;
  return { run };
};

const mapDispatchToProps = dispatch => ({
  getResponse: params => dispatch(getResponse(params))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Display);
