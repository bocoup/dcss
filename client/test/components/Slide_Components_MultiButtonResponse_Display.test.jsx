import React from 'react';
import assert from 'assert';
import {
  fetchImplementation,
  mounter,
  reduxer,
  snapshotter,
  state,
} from '../bootstrap';
import { unmountComponentAtNode } from 'react-dom';
import { mount, render, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Display from '../../components/Slide/Components/MultiButtonResponse/Display.jsx';

const original = JSON.parse(JSON.stringify(state));
let container = null;
let commonProps = null;
let commonState = null;

beforeAll(() => {
  (window || global).fetch = jest.fn();
});

afterAll(() => {
  fetch.mockRestore();
});

beforeEach(() => {
  container = document.createElement('div');
  container.setAttribute('id', 'root');
  document.body.appendChild(container);

  fetchImplementation(fetch);

  commonProps = {
    history: {
      push() {},
    },
  };

  commonState = JSON.parse(JSON.stringify(original));
});

afterEach(() => {
  fetch.mockReset();
  unmountComponentAtNode(container);
  container.remove();
  container = null;
  commonProps = null;
  commonState = null;
});

test('Display', () => {
  expect(Display).toBeDefined();
});

test('Snapshot 1 1', async () => {
  const Component = Display;

  const props = {
    ...commonProps,
    type: 'MultiButtonResponse',
    buttons: [],
    scenario: {
      author: {
        id: 999,
        username: 'owner',
        personalname: 'Owner Account',
        email: 'owner@email.com',
        is_anonymous: false,
        roles: ['participant', 'super_admin', 'facilitator', 'researcher'],
        is_super: true,
      },
      categories: [],
      consent: { id: 57, prose: '' },
      description: 'A Multiplayer Scenario',
      finish: {
        id: 1,
        title: '',
        components: [
          { html: '<h2>Thanks for participating!</h2>', type: 'Text' },
        ],
        is_finish: true,
      },
      lock: {
        scenario_id: 42,
        user_id: 999,
        created_at: '2020-10-10T23:54:19.934Z',
        ended_at: null,
      },
      slides: [
        {
          id: 1,
          title: '',
          components: [
            { html: '<h2>Thanks for participating!</h2>', type: 'Text' },
          ],
          is_finish: true,
        },
        {
          id: 2,
          title: '',
          components: [
            {
              id: 'b7e7a3f1-eb4e-4afa-8569-eb6677358c9e',
              html: '<p>paragraph</p>',
              type: 'Text',
            },
            {
              id: 'aede9380-c7a3-4ef7-add7-838fd5ec854f',
              type: 'TextResponse',
              header: 'TextResponse-1',
              prompt: ',timeout: 0,recallId: ',
              required: true,
              responseId: 'be99fe9b-fa0d-4ab7-8541-1bfd1ef0bf11',
              placeholder: 'Your response',
            },
            {
              id: 'f96ac6de-ac6b-4e06-bd97-d97e12fe72c1',
              html: '<p>?</p>',
              type: 'Text',
            },
          ],
          is_finish: false,
        },
      ],
      status: 1,
      title: 'Multiplayer Scenario',
      users: [
        {
          id: 999,
          email: 'owner@email.com',
          username: 'owner',
          personalname: 'Owner Account',
          roles: ['owner'],
          is_owner: true,
          is_author: true,
          is_reviewer: false,
        },
      ],
      id: 42,
      created_at: '2020-08-31T17:50:28.089Z',
      updated_at: null,
      deleted_at: null,
    },
  };

  const state = {
    ...{},
  };

  const reduxed = reduxer(Component, props, state);
  const mounted = mounter(reduxed, container);
  expect(snapshotter(mounted)).toMatchSnapshot();
});

test('Snapshot 1 2', async () => {
  const Component = Display;

  const props = {
    ...commonProps,
    type: 'MultiButtonResponse',
    buttons: [],
    scenario: {
      author: {
        id: 999,
        username: 'owner',
        personalname: 'Owner Account',
        email: 'owner@email.com',
        is_anonymous: false,
        roles: ['participant', 'super_admin', 'facilitator', 'researcher'],
        is_super: true,
      },
      categories: [],
      consent: { id: 57, prose: '' },
      description: 'A Multiplayer Scenario',
      finish: {
        id: 1,
        title: '',
        components: [
          { html: '<h2>Thanks for participating!</h2>', type: 'Text' },
        ],
        is_finish: true,
      },
      lock: {
        scenario_id: 42,
        user_id: 999,
        created_at: '2020-10-10T23:54:19.934Z',
        ended_at: null,
      },
      slides: [
        {
          id: 1,
          title: '',
          components: [
            { html: '<h2>Thanks for participating!</h2>', type: 'Text' },
          ],
          is_finish: true,
        },
        {
          id: 2,
          title: '',
          components: [
            {
              id: 'b7e7a3f1-eb4e-4afa-8569-eb6677358c9e',
              html: '<p>paragraph</p>',
              type: 'Text',
            },
            {
              id: 'aede9380-c7a3-4ef7-add7-838fd5ec854f',
              type: 'TextResponse',
              header: 'TextResponse-1',
              prompt: ',timeout: 0,recallId: ',
              required: true,
              responseId: 'be99fe9b-fa0d-4ab7-8541-1bfd1ef0bf11',
              placeholder: 'Your response',
            },
            {
              id: 'f96ac6de-ac6b-4e06-bd97-d97e12fe72c1',
              html: '<p>?</p>',
              type: 'Text',
            },
          ],
          is_finish: false,
        },
      ],
      status: 1,
      title: 'Multiplayer Scenario',
      users: [
        {
          id: 999,
          email: 'owner@email.com',
          username: 'owner',
          personalname: 'Owner Account',
          roles: ['owner'],
          is_owner: true,
          is_author: true,
          is_reviewer: false,
        },
      ],
      id: 42,
      created_at: '2020-08-31T17:50:28.089Z',
      updated_at: null,
      deleted_at: null,
    },
  };

  const state = {
    ...commonState,
  };

  const reduxed = reduxer(Component, props, state);
  const mounted = mounter(reduxed, container);
  expect(snapshotter(mounted)).toMatchSnapshot();
});

/*{INJECTION}*/

