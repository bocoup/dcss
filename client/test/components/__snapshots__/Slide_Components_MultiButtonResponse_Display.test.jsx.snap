// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons empty 1`] = `<DocumentFragment />`;

exports[`Buttons undefined 1`] = `<DocumentFragment />`;

exports[`Click a button 1`] = `
<DocumentFragment>
  <div
    class="ui segment"
  >
    <h3
      class="ui header"
      tabindex="0"
    >
      xyz-prompt 
      <div
        aria-label="This prompt requires a response."
        class="ui red floating label prl__label-margin"
      >
        <i
          aria-hidden="true"
          class="asterisk icon prl__icon-margin"
        />
      </div>
    </h3>
    <div
      class="ui list"
      role="list"
    >
      <div
        class="item"
        role="listitem"
      >
        <button
          class="ui fluid button"
          name="xyz-responseId"
          style="background: rgb(255, 0, 255); color: rgb(0, 0, 0);"
          value="Yes"
        >
          Yes
        </button>
      </div>
      <div
        class="item"
        role="listitem"
      >
        <button
          class="ui fluid button"
          name="xyz-responseId"
          style="background: rgb(255, 0, 0); color: rgb(0, 0, 0);"
          value="No"
        >
          No
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Click a button 2`] = `
<DocumentFragment>
  <div
    class="ui segment"
  >
    <h3
      class="ui header"
      tabindex="0"
    >
      xyz-prompt 
      <div
        aria-label="This required prompt has been fulfilled."
        class="ui green floating label prl__label-margin"
      >
        <i
          aria-hidden="true"
          class="check icon prl__icon-margin"
        />
      </div>
    </h3>
    <div
      class="ui list"
      role="list"
    >
      <div
        class="item"
        role="listitem"
      >
        <button
          class="ui fluid button"
          name="xyz-responseId"
          style="background: rgb(255, 0, 255); color: rgb(0, 0, 0);"
          value="Yes"
        >
          <i
            aria-hidden="true"
            class="checkmark icon"
          />
          Yes
        </button>
      </div>
      <div
        class="item"
        role="listitem"
      >
        <button
          class="ui fluid button"
          name="xyz-responseId"
          style="background: rgb(255, 0, 0); color: rgb(0, 0, 0);"
          value="No"
        >
          No
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Click a button, not in run 1`] = `
<DocumentFragment>
  <div
    class="ui segment"
  >
    <h3
      class="ui header"
      tabindex="0"
    >
      xyz-prompt 
      <div
        aria-label="This prompt requires a response."
        class="ui red floating label prl__label-margin"
      >
        <i
          aria-hidden="true"
          class="asterisk icon prl__icon-margin"
        />
      </div>
    </h3>
    <div
      class="ui floating message"
      style="white-space: pre-wrap; overflow-wrap: break-word;"
    >
      <div
        class="content"
      >
        <p>
          Participant response will appear here during scenario run.
        </p>
      </div>
    </div>
    <div
      class="ui list"
      role="list"
    >
      <div
        class="item"
        role="listitem"
      >
        <button
          class="ui fluid button"
          name="xyz-responseId"
          style="background: rgb(255, 0, 255); color: rgb(0, 0, 0);"
          value="Yes"
        >
          Yes
        </button>
      </div>
      <div
        class="item"
        role="listitem"
      >
        <button
          class="ui fluid button"
          name="xyz-responseId"
          style="background: rgb(255, 0, 0); color: rgb(0, 0, 0);"
          value="No"
        >
          No
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Click a button, not in run 2`] = `
<DocumentFragment>
  <div
    class="ui segment"
  >
    <h3
      class="ui header"
      tabindex="0"
    >
      xyz-prompt 
      <div
        aria-label="This prompt requires a response."
        class="ui red floating label prl__label-margin"
      >
        <i
          aria-hidden="true"
          class="asterisk icon prl__icon-margin"
        />
      </div>
    </h3>
    <div
      class="ui floating message"
      style="white-space: pre-wrap; overflow-wrap: break-word;"
    >
      <div
        class="content"
      >
        <p>
          Participant response will appear here during scenario run.
        </p>
      </div>
    </div>
    <div
      class="ui list"
      role="list"
    >
      <div
        class="item"
        role="listitem"
      >
        <button
          class="ui fluid button"
          name="xyz-responseId"
          style="background: rgb(255, 0, 255); color: rgb(0, 0, 0);"
          value="Yes"
        >
          Yes
        </button>
      </div>
      <div
        class="item"
        role="listitem"
      >
        <button
          class="ui fluid button"
          name="xyz-responseId"
          style="background: rgb(255, 0, 0); color: rgb(0, 0, 0);"
          value="No"
        >
          No
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`No color 1`] = `
<DocumentFragment>
  <div
    class="ui segment"
  >
    <h3
      class="ui header"
      tabindex="0"
    >
      xyz-prompt 
      <div
        aria-label="This prompt requires a response."
        class="ui red floating label prl__label-margin"
      >
        <i
          aria-hidden="true"
          class="asterisk icon prl__icon-margin"
        />
      </div>
    </h3>
    <div
      class="ui list"
      role="list"
    >
      <div
        class="item"
        role="listitem"
      >
        <button
          class="ui fluid button"
          name="xyz-responseId"
          value="Yes"
        >
          Yes
        </button>
      </div>
      <div
        class="item"
        role="listitem"
      >
        <button
          class="ui fluid button"
          name="xyz-responseId"
          style="background: rgb(255, 0, 0); color: rgb(0, 0, 0);"
          value="No"
        >
          No
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`No value stored 1`] = `
<DocumentFragment>
  <div
    class="ui segment"
  >
    <h3
      class="ui header"
      tabindex="0"
    >
      xyz-prompt 
      <div
        aria-label="This prompt requires a response."
        class="ui red floating label prl__label-margin"
      >
        <i
          aria-hidden="true"
          class="asterisk icon prl__icon-margin"
        />
      </div>
    </h3>
    <div
      class="ui list"
      role="list"
    >
      <div
        class="item"
        role="listitem"
      >
        <button
          class="ui fluid button"
          name="xyz-responseId"
          style="background: rgb(255, 0, 255); color: rgb(0, 0, 0);"
          value="Yes"
        >
          Yes
        </button>
      </div>
      <div
        class="item"
        role="listitem"
      >
        <button
          class="ui fluid button"
          name="xyz-responseId"
          style="background: rgb(255, 0, 0); color: rgb(0, 0, 0);"
          value="No"
        >
          No
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Previous response value is checked 1`] = `
<DocumentFragment>
  <div
    class="ui segment"
  >
    <h3
      class="ui header"
      tabindex="0"
    >
      xyz-prompt 
      <div
        aria-label="This required prompt has been fulfilled."
        class="ui green floating label prl__label-margin"
      >
        <i
          aria-hidden="true"
          class="check icon prl__icon-margin"
        />
      </div>
    </h3>
    <div
      class="ui list"
      role="list"
    >
      <div
        class="item"
        role="listitem"
      >
        <button
          class="ui fluid button"
          name="xyz-responseId"
          style="background: rgb(255, 0, 255); color: rgb(0, 0, 0);"
          value="Yes"
        >
          <i
            aria-hidden="true"
            class="checkmark icon"
          />
          Yes
        </button>
      </div>
      <div
        class="item"
        role="listitem"
      >
        <button
          class="ui fluid button"
          name="xyz-responseId"
          style="background: rgb(255, 0, 0); color: rgb(0, 0, 0);"
          value="No"
        >
          No
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Previously selected value is checked 1`] = `
<DocumentFragment>
  <div
    class="ui segment"
  >
    <h3
      class="ui header"
      tabindex="0"
    >
      xyz-prompt 
      <div
        aria-label="This required prompt has been fulfilled."
        class="ui green floating label prl__label-margin"
      >
        <i
          aria-hidden="true"
          class="check icon prl__icon-margin"
        />
      </div>
    </h3>
    <div
      class="ui list"
      role="list"
    >
      <div
        class="item"
        role="listitem"
      >
        <button
          class="ui fluid button"
          name="xyz-responseId"
          style="background: rgb(255, 0, 255); color: rgb(0, 0, 0);"
          value="Yes"
        >
          <i
            aria-hidden="true"
            class="checkmark icon"
          />
          Yes
        </button>
      </div>
      <div
        class="item"
        role="listitem"
      >
        <button
          class="ui fluid button"
          name="xyz-responseId"
          style="background: rgb(255, 0, 0); color: rgb(0, 0, 0);"
          value="No"
        >
          No
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Render 1 1 1`] = `
<DocumentFragment>
  <div
    class="ui segment"
  >
    <h3
      class="ui header"
      tabindex="0"
    >
      xyz-prompt 
      <div
        aria-label="This prompt requires a response."
        class="ui red floating label prl__label-margin"
      >
        <i
          aria-hidden="true"
          class="asterisk icon prl__icon-margin"
        />
      </div>
    </h3>
    <div
      class="ui floating message"
      style="white-space: pre-wrap; overflow-wrap: break-word;"
    >
      <div
        class="content"
      >
        <p>
          Participant response will appear here during scenario run.
        </p>
      </div>
    </div>
    <div
      class="ui list"
      role="list"
    >
      <div
        class="item"
        role="listitem"
      >
        <button
          class="ui fluid button"
          name="xyz-responseId"
          style="background: rgb(255, 0, 255); color: rgb(0, 0, 0);"
          value="Yes"
        >
          Yes
        </button>
      </div>
      <div
        class="item"
        role="listitem"
      >
        <button
          class="ui fluid button"
          name="xyz-responseId"
          style="background: rgb(255, 0, 0); color: rgb(0, 0, 0);"
          value="No"
        >
          No
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Render 2 1 1`] = `
<DocumentFragment>
  <div
    class="ui segment"
  >
    <h3
      class="ui header"
      tabindex="0"
    >
      xyz-prompt 
      <div
        aria-label="This prompt requires a response."
        class="ui red floating label prl__label-margin"
      >
        <i
          aria-hidden="true"
          class="asterisk icon prl__icon-margin"
        />
      </div>
    </h3>
    <div
      class="ui list"
      role="list"
    >
      <div
        class="item"
        role="listitem"
      >
        <button
          class="ui fluid button"
          name="xyz-responseId"
          style="background: rgb(255, 0, 255); color: rgb(0, 0, 0);"
          value="Yes"
        >
          Yes
        </button>
      </div>
      <div
        class="item"
        role="listitem"
      >
        <button
          class="ui fluid button"
          name="xyz-responseId"
          style="background: rgb(255, 0, 0); color: rgb(0, 0, 0);"
          value="No"
        >
          No
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
